src/scripts/index.ts

```diff
@@ -32,17 +34,53 @@ const getMajorCities: NodeListOf<HTMLElement> =
 const majorCities: HTMLElement[] = Array.from(getMajorCities);
 
 for (const city of majorCities) {
-  city.addEventListener("click", function (e) {
-    getCityCoords(city.innerText);
+  city.addEventListener("click", function () {
+    console.log("this is running");
+    const cityData = getCityCoords(city.innerText);
+
+    if (Object.keys(cityData).length === 0) {
+      throw new Error(`No data for ${city.innerText}`);
+    }
+    weatherFetchRequest$.next(cityData);
+    subscribeToWeatherAPI(cityData);
```

Comment:
Nope, you can't rebuild the marble run/hotwheels™ track every time you put a marble/car on the track